frontend/src/util/activity.js

```diff
@@ -51,35 +53,42 @@ const createContentActivity = async (activityParams, messageList, apiUrl) => {
 
   let content = newestMessage.fields.content
 
-  if (content.content_type === TLM_ST.COMMENT ||
-    content.content_type === TLM_ST.MENTION ||
-    content.content_type === TLM_ST.TODO) {
+  const isMentionOrComment = content.content_type === TLM_ST.COMMENT ||
+    content.content_type === TLM_ET.MENTION
+
+  let parentContentType
+  if (isMentionOrComment) parentContentType = content.parent_content_type
+  else if (content.assignee) parentContentType = content.parent.content_type
+  else if (content.parent_id) {
+    const parentType = await getParentType(content.parent_id, apiUrl)
+    if (parentType !== CONTENT_TYPE.FOLDER) parentContentType = parentType
+  }
+
+  let fetchGetWorkspaceContent
+  if (parentContentType) {
     // INFO - SG - 2021-04-16
     // We have to get the parent content as comments shall produce an activity
     // for it and not for the comment.
-    const parentContentType = content.parent_content_type
-    const parentId = content.parent_id
+    const parentId = content.parent_id || content.parent.content_id
     if (!(parentContentType && parentId)) return null
-
-    const response = await handleFetchResult(await getWorkspaceContent(
+    fetchGetWorkspaceContent = await handleFetchResult(await getWorkspaceContent(
       apiUrl,
       newestMessage.fields.workspace.workspace_id,
       parentContentType,
       parentId
     ))
-    if (!response.apiResponse.ok) return null
-    content = { ...content, ...response.body }
   } else {
-    const response = await handleFetchResult(await getWorkspaceContent(
+    fetchGetWorkspaceContent = await handleFetchResult(await getWorkspaceContent(
       apiUrl,
       newestMessage.fields.workspace.workspace_id,
       content.content_type,
       content.content_id
     ))
-    if (!response.apiResponse.ok) return null
-    content = { ...content, ...response.body }
   }
```

Comment:
I think you can simplify this part like so:
```jsx
    if (!(parentContentType && parentId)) return null
    const fetchGetWorkspaceContent = await handleFetchResult(await getWorkspaceContent(
      apiUrl,
      newestMessage.fields.workspace.workspace_id,
      parentContentType ? parentContentType : content.content_type,
      parentContentType ? parentId : content.content_id
    ))
```

Reply:
Fixed